mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -40,25 +41,44 @@
 import reactor.core.scheduler.Schedulers;
 
 public class InVmEventDelivery implements EventDelivery {
+
+    private enum DeliveryOption {
+        NO_RETRY, WITH_RETRY
+    }
+
     private static final Logger LOGGER = LoggerFactory.getLogger(InVmEventDelivery.class);
-    private static final int MAX_RETRIES = 3;
-    private static final Duration FIRST_BACKOFF = Duration.ofMillis(100);
     private static final Duration MAX_BACKOFF = Duration.ofMillis(Long.MAX_VALUE);
-    private static final double DEFAULT_JITTER_FACTOR = 0.5;
 
     private final MetricFactory metricFactory;
+    private final RetryBackoffConfiguration retryBackoff;
 
-    @Inject
```

Comment:
Can we keep injecting this in Guice? + bind it there to the default instance.

You can call the other constructor in Spring by writing stuff in `mailbox/spring/src/main/resources/META-INF/spring/event-system.xml`

Reply:
Spring works well with two `@Injected` constructors, only Guice has problem, it doesn't recognize good constructor to go.

Reply:
No you keep the @Inject on the one you want to use  with Guice as Spring config don't need @Inject